aptos-move/block-executor/src/index_mapping.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright © Aptos Foundation
+
+use aptos_mvhashmap::types::TxnIndex;
+
+/// In standard BlockSTM, the internal states assume contiguous transactions indices.
+/// To also support sharded execution (where each shard gets non-contiguous transaction indices
+/// and possibly waits for transaction results from other shards),
+/// an index mapping is need as an additional input to our existing BlockSTM implementation.
+#[derive(Clone)]
+pub struct IndexMapping {
+    /// A sorted list of transaction indices.
+    indices: Vec<usize>,
+    /// A TxnIndex -> local position mapping.
+    /// Currently implemented as a `Vec` of size equal to the block size, assuming it's not too large.
+    positions_by_index: Vec<usize>,
+}
+
+impl IndexMapping {
+    /// Create Positions by TxnIndex mapping from the TxnIndex list.
+    pub fn inverse(block_size: usize, indices: &[usize]) -> Vec<usize> {
+        let mut ret = vec![usize::MAX; block_size];
+        for (pos, &index) in indices.iter().enumerate() {
+            ret[index] = pos;
+        }
+        ret
+    }
+
+    pub fn new(indices: Vec<usize>, block_size: usize) -> Self {
+        let positions_by_index = Self::inverse(block_size, &indices);
+        Self {
+            indices,
+            positions_by_index,
+        }
+    }
+
+    pub fn wrap(indices: Vec<usize>, positions_by_idx: Vec<usize>) -> Self {
```

Comment:
It doesn't seem this is being used anywhere?

Reply:
fixed